python/lsst/pipe/tasks/selectImages.py

```diff
@@ -590,3 +591,119 @@ def run(self, wcsList, bboxList, coordList, psfList, dataIds, **kwargs):
                       f"range of {psfSizes[indices.index(output[0])]}"
                       f"--{psfSizes[indices.index(output[-1])]} arcseconds")
         return output
+
+
+class BestSeeingSelectVisitsConnections(pipeBase.PipelineTaskConnections,
+                                        dimensions=("tract", "patch", "skymap", "band", "instrument"),
+                                        defaultTemplates={"coaddName": "bestSeeing"}):
+    sourceTables = pipeBase.connectionTypes.Input(
+        doc="Source table with PSF size information",
+        name="sourceTable_visit",
+        storageClass="DataFrame",
+        dimensions=("instrument", "visit"),
+        multiple=True,
+        deferLoad=True
+    )
+    goodVisits = pipeBase.connectionTypes.Output(
+        doc="Selected visits to be coadded.",
+        name="{coaddName}VisitsDict",
+        storageClass="StructuredDataDict",
+        dimensions=("instrument", "tract", "patch", "skymap", "band"),
+    )
+
+
+class BestSeeingSelectVisitsConfig(pipeBase.PipelineTaskConfig,
+                                   pipelineConnections=BestSeeingSelectVisitsConnections):
+    nImagesMax = pexConfig.RangeField(
+        dtype=int,
+        doc="Maximum number of images to select",
+        default=5,
+        min=0)
+    maxPsfFwhm = pexConfig.Field(
+        dtype=float,
+        doc="Maximum PSF FWHM (in arcseconds) to select",
+        default=1.5,
+        optional=True)
+    minPsfFwhm = pexConfig.Field(
+        dtype=float,
+        doc="Minimum PSF FWHM (in arcseconds) to select",
+        default=0.,
+        optional=True)
+
+
+class BestSeeingSelectVisitsTask(pipeBase.PipelineTask):
+    ConfigClass = BestSeeingSelectVisitsConfig
+    _DefaultName = 'bestSeeingSelectVisits'
+
+    def run(self, sourceTables):
+        inputVisits = [sourceTable.ref.dataId['visit'] for sourceTable in sourceTables]
+        psfSizes = []
+        visits = []
+        for visit, sourceTable in zip(inputVisits, sourceTables):
+            df = sourceTable.get(parameters={"columns": ['IxxPsf', 'IyyPsf', 'IxyPsf',
+                                                         'LocalWcs_CDMatrix_2_1', 'LocalWcs_CDMatrix_1_1',
+                                                         'LocalWcs_CDMatrix_1_2', 'LocalWcs_CDMatrix_2_2']})
+            wcsDet = (df.LocalWcs_CDMatrix_1_1 * df.LocalWcs_CDMatrix_2_2
+                      - df.LocalWcs_CDMatrix_1_2 * df.LocalWcs_CDMatrix_1_2)
+            # if min/max PSF values are defined, remove images out of bounds
+            pixToArcseconds = np.nanmedian(3600*np.degrees(np.sqrt(np.fabs(wcsDet))))
+            determinantRadius = np.power(np.nanmedian(df.IxxPsf * df.IxxPsf - df.IxyPsf**2), 0.25)
```

Comment:
I've never used the `LocalWcs_CDMatrix` columns before! I'd feel safer (less error-prone) getting each visit's WCS and PSF directly (via `calexp.wcs` and `calexp.psf`) and using build-in methods to access the pixel scale and PSF size. Did you do it this way because it's faster?